pkg/agent/plugin/workloadattestor/k8s/k8s_posix.go

```diff
@@ -175,3 +211,25 @@ func canonicalizePodUID(uid string) types.UID {
 		return r
 	}, uid))
 }
+
+func configureSigstoreClient(client sigstore.Sigstore, c *SigstoreHCLConfig, log hclog.Logger) error {
+	// Configure sigstore settings
+	client.ClearSkipList()
+	if c.SkippedImages != nil {
+		client.AddSkippedImage(c.SkippedImages)
+	}
+	client.EnableAllowSubjectList(c.AllowedSubjectListEnabled)
+	client.SetLogger(log)
+	client.ClearAllowedSubjects()
+	for _, subject := range c.AllowedSubjects {
+		client.AddAllowedSubject(subject)
+	}
+	rekorURL := "https://rekor.sigstore.dev/" // default rekor url
+	if c.RekorURL != nil {
+		rekorURL = (*c.RekorURL)
+	}
```

Comment:
Is it necessary to include this default URL in the compiled binary? While one is able to to override that value, if it's not overridden (on purpose or by mistake) then in-use images will become public knowledge. The act of pulling image signatures from https://rekor.sigstore.dev has the add-on effect of telling https://rekor.sigstore.dev  that a particular `image#tag` is in use. That information can be used to inform multiple attack vectors.

I would prefer if this URL is a required configuration value with no default. This minor change will cause every organization using Sigstore during workload attestation to make an informed decision about this setting, and will make it apparent when there are calls being made to an external URL. Some organizations will choose to not use the public endpoint and will instead stand up their own instance. But by making that a necessary choice it forces the conversation and decision.

As coded now, those outgoing calls may come as an unintended surprise.